### DISCUSSION TITLE
SM2签名验签功能中，公私钥，只能是PKCS8结构吗，能不能支持16进制的裸公钥和裸私钥

### DISCUSSION THREAD

**Discussion by walle-r** [Leon406/ToolsFx#270] [Ideas]
![image](https://github.com/Leon406/ToolsFx/assets/48503060/147d93ff-90f3-4db8-92cc-9e74ea480cb1)

**Comment by Leon406** [comment]
不支持，还是按照标准来吧

**Reply by walle-r** [reply]
因为工作中收到是很多都是裸公私钥，有什么简单的办法或工具能转成PKCS8格式吗

**Reply by Leon406** [reply]
提供下完整示例，有空的话研究下

**Reply by TxiRan** [reply]
SM2公钥：`0491762dd69d4035eede32d88b95fc290c8d8cc8a51901c55f6cd2b4e5c1d654e53c0d0f7ea93c5208f7acabb4c1f966ac0fc9c67f5986c3ce44821e1b50010c5b`
SM2私钥：`cf34d040188892a6487a3db52fe29aa63e3932f090d38abbed53782b8d231902`



#### 私钥加签

SM2私钥：`cf34d040188892a6487a3db52fe29aa63e3932f090d38abbed53782b8d231902`
待签名内容：`ABCD`
对`ABCD`签名结果：`30450220347e2d4ed185b7482a8db297ef46edbefe4a9a11bf4d49de07b7a034959e69d5022100ce7cbcef92d832d22a0681172e77ac757840ac24b085be78d1cf8b12d459350c`

<img width="735" height="872" alt="image" src="https://github.com/user-attachments/assets/9b893def-adf9-4331-8aae-945eefdabec2" />


#### 公钥验签

SM2公钥：`0491762dd69d4035eede32d88b95fc290c8d8cc8a51901c55f6cd2b4e5c1d654e53c0d0f7ea93c5208f7acabb4c1f966ac0fc9c67f5986c3ce44821e1b50010c5b`
待验签内容：`ABCD`
对`ABCD`签名结果：`30450220347e2d4ed185b7482a8db297ef46edbefe4a9a11bf4d49de07b7a034959e69d5022100ce7cbcef92d832d22a0681172e77ac757840ac24b085be78d1cf8b12d459350c`
验签结果：`签名一致`

<img width="735" height="872" alt="image" src="https://github.com/user-attachments/assets/3a3b8860-bfe5-446a-b078-d1ce1b7a5857" />

#### ToolsFx当前结果
<img width="1075" height="872" alt="image" src="https://github.com/user-attachments/assets/dc2bb055-9cef-47f5-b405-8bbd411a1b38" />

<img width="1071" height="827" alt="image" src="https://github.com/user-attachments/assets/f45c8cf5-9dfa-4652-b122-c185e6fffdd1" />

SM2新国标中的签名验签功能是否有变动
